experiments/list_unsupported.sh

```diff
@@ -145,213 +67,12 @@ if [ -z "$project_dir" ]; then
-      awk '/^ *[0-9][0-9]* .*/ { \
-         print "--------------------------------------------------------------------------------"; \
-         print "Occurs:", $1, "times"; \
-         sub(/^ *[0-9][0-9]* */,"",$0); \
-         gsub(/<<<>>>/,"\n", $0); \
-         print $0; \
-         print "--------------------------------------------------------------------------------"; \
-      }'
-
-# For exceptions - note that we don't redact the file name here, because the filename/location
-# will be one of our own gnat2goto or gnat source files, not the users code.
-awk  '/^raised .*$/ { \
-         if (prev == "<========================>") \
-            print prev "<<<>>>" $0 "<<<>>>" ; \
-      } \
-      { \
-         prev = $0 \
-      }' "$file_name".txt \
-   | LC_ALL=posix sort | uniq -c | LC_ALL=posix sort -k1,1nr -k2 | \
-      awk '/^ *[0-9][0-9]* .*/ { \
-         print "--------------------------------------------------------------------------------"; \
-         print "Occurs:", $1, "times"; \
-         sub(/^ *[0-9][0-9]* */,"",$0); \
-         gsub(/<<<>>>/,"\n", $0); \
-         print $0; \
-         print "--------------------------------------------------------------------------------"; \
-      }'
-
-# For any other kind of failure, we will have logged the exit code. To summarize
-# these cases, we convert the log output into a single line 'canonical form'
-# (replace newlines with '<<<>>>>') then we can sort and uniq the list, before
-# then turning the resulting list back into the same format as the report above.
-# Yes, it's ugly...
-awk  '/^---------- COMPILING: / { \
-         buf = ""; count=0; \
-      } \
-      /^gnat2goto exit code: [0-9]*/ { \
-         sub(/^.*---------- COMPILING: [^<]*<<<>>>/,"",buf); \
-         print buf "<<<>>>" $0; \
-      } \
-      { \
-         buf = buf "<<<>>>" $0 \
-      }' "$file_name".txt \
-   | LC_ALL=posix sort | uniq -c | LC_ALL=posix sort -k1,1nr -k2 | \
-      awk '/^ *[0-9][0-9]* .*/ { \
-         print "--------------------------------------------------------------------------------"; \
-         print "Occurs:", $1, "times"; \
-         sub(/^ *[0-9][0-9]* */,"",$0); \
-         gsub(/<<<>>>/,"\n", $0); \
-         print $0; \
-         print "--------------------------------------------------------------------------------"; \
-      }'
-
-# Gather overall statistics
-total_count=`grep -c -E '^---------- COMPILING: ' "$file_name".txt`
-fail_count=`grep -c -E '^---------- FAILED ' "$file_name".txt`
-non_compile_error_fail_count=`grep -c -E '^gnat2goto exit code: [0-9]*' "$file_name".txt`
-missing_feature_count=`grep -c -E '^---------- MISSING FEATURES ' "$file_name".txt`
-ok_count=`grep -c -E '^---------- OK ' "$file_name".txt`
+# Invoke the building script on project
+. "$(dirname $0)/list_unsupported_build.sh"
 
+# Invoke the report generating script on project
+. "$(dirname $0)/list_unsupported_report.sh" "$file_name".stdout "$file_name".stdout
```

Comment:
This still makes the implicit assumption that the `list_unsupported_build.sh` knows it needs to generate `"$file_name".stdout`, etc. I'd much rather just see:
`. "$(dirname $0)/list_unsupported_build.sh" > "blah.stdout" 2> "blah.stderr"`